xlators/storage/posix/src/posix-entry-ops.c

```diff
@@ -215,6 +220,28 @@ posix_lookup(call_frame_t *frame, xlator_t *this, loc_t *loc, dict_t *xdata)
     op_ret = -1;
     if (gf_uuid_is_null(loc->pargfid) || (loc->name == NULL)) {
         /* nameless lookup */
+        op_ret = op_errno = 0;
+        /* The gfid will be renamed to ".glusterfs/unlink" in case
+         * there are any open fds on the file in posix_unlink path.
+         * So client can request server to do nameless lookup with
+         * xdata = GF_NAMELESS_UNLINK_DIR_LOOKUP in ".glusterfs/unlink"
+         * dir if a client wants to know the status of the all open fds
+         * on the unlinked file. If the file still present in the
+         * ".glusterfs/unlink" dir then it indicates there still
+         * open fds present on the file and the file is still under
+         * unlink process */
+        ret = dict_get_uint32(xdata, GF_UNLINKED_LOOKUP, &lookup_unlink_dir);
+        if (!ret && lookup_unlink_dir) {
+            POSIX_GET_FILE_UNLINK_PATH(priv->base_path, loc->gfid, unlink_path);
+            ret = sys_lstat(unlink_path, &lstatbuf);
```

Comment:
Probably it would be better to use `buf` instead of `lstatbuf` so that we return a valid iatt if the file still exists. It could be useful for some cases. In this case we should set `ia_nlink = 0` to be consistent.

Reply:
My bad. Changing lstatbuf to buf. Btw, why "ia_nlink = 0" should be done when the file exists?

Reply:
Please correct me if wrong. sys_lstat() wants struct stat which lstatbuf as 2nd arg and buf is of struct attr ?. What should be replaced here? The only intention here is to verify the file presence under unlink dir and should we construct iattr if file present under unlink dir ?

Also, in what case ia_nlink = 0 should be done ?

Reply:
If a client wants to lookup a file, it might be to use it. Since the file still exists, I think we should return all the information we can, at least the iatt. It's true that `buf` has a different type. We should convert `lstatbuf` to `struct iatt` in `buf` to return it.

I'm also not sure if it wouldn't be better to also return xattrs as regular lookups do, but for now I think iatt data is enough.

`ia_nlink` should always be set to 0, as the file is deleted so it doesn't have any link.

Reply:
Okay. will do iatt_from_stat() and set ia_nlink to 0 in case of ret == ENOENT